clash-prelude/src/Clash/Sized/Vector.hs

```diff
@@ -208,9 +210,10 @@ let populationCount' :: (KnownNat k, KnownNat (2^k)) => BitVector (2^k) -> Index
 --   ending at @'length' - 1@.
 data Vec :: Nat -> Type -> Type where
   Nil  :: Vec 0 a
-  Cons :: a -> Vec n a -> Vec (n + 1) a
+  (:>) :: a -> Vec n a -> Vec (n + 1) a
 
-infixr 5 `Cons`
+infixr 5 :>
+{-# COMPLETE (:>) #-}
```

Comment:
No. I don't know why we ever had

```
{-# COMPLETE (:>)       #-}	
{-# COMPLETE Nil, (:>)  #-}
```

in there previously, that obviously slipped through the review. I'd rather have the "false positive" behavior of the warnings GHC emits, than the "false negatives" this COMPLETE pragma gives you. Also in GHC 8.12 (maybe even 8.10), the pattern match checker is much improved, hopefully no longer giving the "false positives".

Reply:
I fully agree! Good to know that future GHC's will do it better.

Reply:
Can I read about the improved pattern match checker? I've found [this](https://gitlab.haskell.org/ghc/ghc/wikis/Migration/8.10#improvements-to-pattern-match-checking-with-user-defined-complete-sets), but that doesn't seem relevant for us.

Reply:
SPJ gave a keynote about it at Haskell Exchange this year